disnake/interactions/base.py

```diff
@@ -682,13 +682,21 @@ class InteractionResponse:
     """
 
     __slots__: Tuple[str, ...] = (
-        "_responded",
         "_parent",
+        "_response_type",
     )
 
     def __init__(self, parent: Interaction):
         self._parent: Interaction = parent
-        self._responded: bool = False
+        self._response_type: Optional[InteractionResponseType] = None
+
+    @property
+    def response_type(self) -> Optional[InteractionResponseType]:
```

Comment:
```suggestion
    def type(self) -> Optional[InteractionResponseType]:
```
![image](https://user-images.githubusercontent.com/8530778/189687165-cb90e89d-b717-4220-90fe-b421f311792e.png)